main/gradio_ui/builder.py

```
@@ -0,0 +1,208 @@
+
+        async def on_stop_click():
+            global auto_stop_task, stop_requested
+            # 1) Immediate UI update
+            yield gr.update(value="Stopped", interactive=False)
+            # 2) Cancel timers & background loop
+            stop_requested = True
+            # Signal cooperative stop to trading loop
+            try:
+                request_stop()
+            except Exception:
+                pass
```

Ruff (0.13.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)